exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -491,259 +648,585 @@ func (e *Exchange) SubmitOrder(ctx context.Context, s *order.Submit) (*order.Sub
 		Status: make(map[string]string),
 	}
-	openOrders, err := e.GetOpenOrdersForAllCurrencies(ctx)
-	if err != nil {
-		return cancelAllOrdersResponse, err
+	if cancelOrd == nil {
+		return cancelAllOrdersResponse, common.ErrNilPointer
 	}
-
-	for key := range openOrders.Data {
-		for i := range openOrders.Data[key] {
-			err = e.CancelExistingOrder(ctx, openOrders.Data[key][i].OrderNumber)
+	var pairs currency.Pairs
+	if !cancelOrd.Pair.IsEmpty() {
+		fPair, err := e.FormatExchangeCurrency(cancelOrd.Pair, cancelOrd.AssetType)
+		if err != nil {
+			return cancelAllOrdersResponse, err
+		}
+		pairs = append(pairs, fPair)
+	}
+	switch cancelOrd.AssetType {
+	case asset.Spot:
+		switch cancelOrd.Type {
+		case order.TrailingStop, order.TrailingStopLimit, order.StopLimit, order.Stop:
+			var orderTypes []OrderType
+			if cancelOrd.Type != order.UnknownType {
+				orderTypes = append(orderTypes, OrderType(cancelOrd.Type))
+			}
+			resp, err := e.CancelSmartOrders(ctx, pairs, nil, orderTypes)
 			if err != nil {
-				id := strconv.FormatInt(openOrders.Data[key][i].OrderNumber, 10)
-				cancelAllOrdersResponse.Status[id] = err.Error()
+				return cancelAllOrdersResponse, err
 			}
+			for _, co := range resp {
+				cancelAllOrdersResponse.Status[co.OrderID] = co.State
+			}
+		default:
+			if e.Websocket.IsConnected() && e.Websocket.CanUseAuthenticatedEndpoints() && e.Websocket.CanUseAuthenticatedWebsocketForWrapper() {
+				wsResponse, err := e.WsCancelTradeOrders(ctx, pairs.Strings(), []AccountType{AccountType(cancelOrd.AssetType)})
+				if err != nil {
+					return cancelAllOrdersResponse, err
+				}
+				for _, wco := range wsResponse {
+					if wco.Code == 0 {
+						cancelAllOrdersResponse.Status[strconv.FormatUint(wco.OrderID, 10)] = wco.State
+					}
+				}
+			} else {
+				resp, err := e.CancelTradeOrders(ctx, pairs.Strings(), []AccountType{AccountType(cancelOrd.AssetType)})
+				if err != nil {
+					return cancelAllOrdersResponse, err
+				}
+				for _, co := range resp {
+					if co.Code == 0 {
+						cancelAllOrdersResponse.Status[co.OrderID] = co.State
+					}
+				}
+			}
+		}
+	case asset.Futures:
+		result, err := e.CancelFuturesOrders(ctx, cancelOrd.Pair.String(), cancelOrd.Side.String())
+		if err != nil {
+			return cancelAllOrdersResponse, err
 		}
+		for _, co := range result {
+			if co.Code == 200 {
+				cancelAllOrdersResponse.Status[co.OrderID] = "cancelled"
+			}
+		}
+	default:
+		return cancelAllOrdersResponse, fmt.Errorf("%w: %q", asset.ErrNotSupported, cancelOrd.AssetType)
 	}
-
 	return cancelAllOrdersResponse, nil
 }
 
 // GetOrderInfo returns order information based on order ID
-func (e *Exchange) GetOrderInfo(ctx context.Context, orderID string, pair currency.Pair, _ asset.Item) (*order.Detail, error) {
+func (e *Exchange) GetOrderInfo(ctx context.Context, orderID string, pair currency.Pair, assetType asset.Item) (*order.Detail, error) {
 	if pair.IsEmpty() {
 		return nil, currency.ErrCurrencyPairEmpty
 	}
```

Comment:
🚧 Change request:
* Since this exchange doesn't require a currency pair, I'm not sure we should be requiring that it's not empty.
Some exchanges enforce it, but not all.
* L1052 we do checks as if it's okay for pair to be empty
* If the order isn't a smart order we don't validate the pair matches the order at all 🤷‍♂️ 
* smartOrder.Symbol isn't a currency Symbol, but the futures one is
* Making it a symbol simplifies code in a few other places, included in patch

So I'd suggest we just add a comment and then simplify: gbjk@b15b839c7
```
curl -s https://github.com/gbjk/gocryptotrader/commit/b15b839c7.patch | git am
```

Reply:
🚧 Change request:
This patch was partially applied.
Was there issues with it?
We still have the inconsistency in checking Pair - smart orders do, and normal orders don't.

I still say:
Just remove the pair is empty check at the top of `GetOrderInfo`, and remove the verification that they're the same.
Add a comment to the method

**If you don't want to do that, just say why, but you have to fix the inconsistency in non-smart orders then**